packages/node/src/utils/blockSizeBuffer.ts

```diff
@@ -0,0 +1,34 @@
+// Copyright 2020-2021 OnFinality Limited authors & contributors
+// SPDX-License-Identifier: Apache-2.0
+
+import { Queue } from '@subql/node-core';
+
+export class BlockSizeBuffer extends Queue<number> {
+  constructor(capacity: number) {
+    super(capacity);
+  }
+
+  putMany(items: number[]): void {
+    if (items.length > this.capacity) {
+      throw new Error(`Queue out of capacity`);
+    }
+
+    if (this.capacity && items.length > this.freeSpace) {
+      this.takeMany(items.length);
```

Comment:
Why does putting items remove items from the queue? This is very implicit and items would seem to disappear

Reply:
we only need the latest block sizes to compute moving average, so we need to make space for the new items by removing the old ones